components/suggestion/search_suggestion_list.jsx

```diff
@@ -96,18 +120,7 @@ export default class SearchSuggestionList extends SuggestionList {
             }
 
             if (isSelection) {
-                if (item.type === Constants.DM_CHANNEL || item.type === Constants.GM_CHANNEL) {
-                    this.currentLabel = item.display_name;
-                } else if (item.username) {
-                    this.currentLabel = item.username;
-                } else {
-                    this.currentLabel = item.name;
-                }
-
-                // Pause the event loop and Wait for the aria-live element to be up
-                setTimeout(() => {
-                    this.announceLabel();
-                }, Constants.OVERLAY_TIME_DELAY_SMALL);
+                this.generateLabel(item);
```

Comment:
This should be done outside of the render method since the render method is supposed to have no side effects. Could you move this to `componentDidUpdate`? Since you'll have to get the selection again, you could have
```javascript
componentDidUpdate(prevProps) {
    if (this.props.selection !== prevProps.selection) {
        this.generateLabel(this.props.items.find((item) => item.term === this.props.selection));
    }
}
```
That would also let you remove the `setTimeout` since it would be called after the DOM is updated

Reply:
Done, but still had to use the `setTimeout`.